### ISSUE TITLE
User guide to use Rematerialization

### ISSUE THREAD

**Issue by pure-rgb** [keras-team/keras#20964]
gradient checkpoint feat was introduced here https://github.com/keras-team/keras/pull/20743 but currently there is no user guide how to adop it or use it.

**Comment by pure-rgb** [comment]
what? keras-team-review-pending tag is added for such issue, seriously! lol.

**Comment by pure-rgb** [comment]
Summary:
1. How to leverage gradient-checkpoint for the keras.application models: https://keras.io/api/applications/
2. How to for keras-hub model
3. How to for custom model, or user-define model
4. Simple usage. And how to adopt for complicated model like YOLO-v8. 
5. Will it work for all backend out of the box? Any comparison are made for gradient checkpoint among backends? which backends performs well?
6. State of the limitation of keras.remat?

**Comment by sonali-kumari1** [comment]
Hi @pure-rgb - 

Please refer to this documentation on [remat](https://keras.io/api/rematerialization/remat/) and [RematScope](https://keras.io/api/rematerialization/remat_scope/). Thanks!
